src/utils/init/config-github.js

```diff
@@ -1,188 +1,215 @@
-const fs = require('fs')
-const os = require('os')
-const path = require('path')
-
 const { Octokit } = require('@octokit/rest')
 const chalk = require('chalk')
-const inquirer = require('inquirer')
-const parseGitRemote = require('parse-github-url')
 
-const { version } = require('../../../package.json')
 const ghauth = require('../gh-auth')
 
-const { makeNetlifyTOMLtemplate } = require('./netlify-toml-template')
+const { getBuildSettings, saveNetlifyToml, formatErrorMessage, createDeployKey, updateSite } = require('./utils')
 
-const UA = `Netlify CLI ${version}`
+const formatRepoAndOwner = ({ repoOwner, repoName }) => {
+  return {
+    name: chalk.magenta(repoName),
+    owner: chalk.magenta(repoOwner),
+  }
+}
 
 const PAGE_SIZE = 100
 
-module.exports = async function configGithub(ctx, site, repo) {
-  const { api, globalConfig } = ctx.netlify
-  const current = globalConfig.get('userId')
+const isValidToken = (token) => {
+  return token && token.user && token.token
+}
 
-  let ghtoken = globalConfig.get(`users.${current}.auth.github`)
+const getGitHubToken = async ({ log, globalConfig }) => {
+  const userId = globalConfig.get('userId')
+  const githubToken = globalConfig.get(`users.${userId}.auth.github`)
 
-  if (!ghtoken || !ghtoken.user || !ghtoken.token) {
-    const newToken = await ghauth({
-      opts: {
-        scopes: ['admin:org', 'admin:public_key', 'repo', 'user'],
-        userAgent: UA,
-        note: `Netlify CLI ${os.userInfo().username}@${os.hostname()}`,
-      },
-      log: ctx.log,
-    })
-    globalConfig.set(`users.${current}.auth.github`, newToken)
-    ghtoken = newToken
+  if (isValidToken(githubToken)) {
+    return githubToken.token
   }
-  const octokit = new Octokit({
-    auth: `token ${ghtoken.token}`,
-  })
-
-  const key = await api.createDeployKey()
-  const parsedURL = parseGitRemote(repo.repo_path)
 
-  await octokit.repos.addDeployKey({
-    title: 'Netlify Deploy Key',
-    key: key.public_key,
-    repo: parsedURL.name,
-    owner: parsedURL.owner,
-    read_only: true,
+  const newToken = await ghauth({
+    log,
   })
+  globalConfig.set(`users.${userId}.auth.github`, newToken)
+  return newToken.token
+}
 
-  repo.deploy_key_id = key.id
-
-  // TODO: Look these up and default to the lookup order
+const getGitHubClient = ({ token }) => {
+  const octokit = new Octokit({
+    auth: `token ${token}`,
+  })
+  return octokit
+}
 
-  let defaultBuildCmd
```

Comment:
Done in `getBuildSettings`.

Also there is a bug here - `publish` and `functions` are absolute paths (if they exist) so they need to be normalized to a relative path before using them as default values